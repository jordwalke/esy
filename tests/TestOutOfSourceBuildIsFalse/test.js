const {exec: execBase, readFile} = require('../harness');

function exec(cmd) {
  return execBase(cmd, {cwd: __dirname});
}

test('env', () => {
  let res = exec('../../.bin/esy');
  expect(res.status).toBe(0);
  expect(res.stdout.toString()).toMatchSnapshot();
});

test('forces build to happen in $cur__target_dir', () => {
  let res = exec(`
  rm -rf _build _install _esy_store
  ../../.bin/esy build
  `);
  expect(res.status).toBe(0);
  expect(readFile(__dirname, 'OK')).toBe('OK');
  expect(readFile(__dirname, '_build', 'OK')).toBe('OK!!!');
});
