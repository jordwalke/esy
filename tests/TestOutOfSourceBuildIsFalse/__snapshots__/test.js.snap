exports[`test env 1`] = `
"
# EsySandboxVariables /Users/andreypopp/Workspace/esy/tests/TestOutOfSourceBuildIsFalse
export PATH=\"/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin\"
export SHELL=\"env -i /bin/bash --norc --noprofile\"
export esy__platform=\"platform\"
export esy__architecture=\"architecture\"
export esy__target_platform=\"platform\"
export esy__target_architecture=\"architecture\"
# [BuiltIns]
export esy__sandbox=\"$ESY__SANDBOX\"
export esy__store=\"$ESY__STORE\"
export esy__install_tree=\"$esy__sandbox/_install\"
export esy__build_tree=\"$esy__sandbox/_build\"
export cur__name=\"TestOutOfSourceBuildIsFalse\"
export cur__version=\"1.0.0\"
export cur__source_root=\"$esy__sandbox\"
export cur__root=\"$esy__sandbox/_build\"
export cur__depends=\"\"
export cur__target_dir=\"$esy__sandbox/_build\"
export cur__install=\"$esy__sandbox/_install\"
export cur__bin=\"$cur__install/bin\"
export cur__sbin=\"$cur__install/sbin\"
export cur__lib=\"$cur__install/lib\"
export cur__man=\"$cur__install/man\"
export cur__doc=\"$cur__install/doc\"
export cur__stublibs=\"$cur__install/stublibs\"
export cur__toplevel=\"$cur__install/toplevel\"
export cur__share=\"$cur__install/share\"
export cur__etc=\"$cur__install/etc\"
# [Custom Variables]
export OCAMLFIND_CONF=\"$cur__target_dir/_esy_findlib.conf\"

# TestOutOfSourceBuildIsFalse@1.0.0 $esy__sandbox/package.json
# [BuiltIns]
export testoutofsourcebuildisfalse__name=\"TestOutOfSourceBuildIsFalse\"
export testoutofsourcebuildisfalse__version=\"1.0.0\"
export testoutofsourcebuildisfalse__source_root=\"$esy__sandbox\"
export testoutofsourcebuildisfalse__root=\"$esy__sandbox/_build\"
export testoutofsourcebuildisfalse__depends=\"\"
export testoutofsourcebuildisfalse__target_dir=\"$esy__sandbox/_build\"
export testoutofsourcebuildisfalse__install=\"$esy__sandbox/_install\"
export testoutofsourcebuildisfalse__bin=\"$testoutofsourcebuildisfalse__install/bin\"
export testoutofsourcebuildisfalse__sbin=\"$testoutofsourcebuildisfalse__install/sbin\"
export testoutofsourcebuildisfalse__lib=\"$testoutofsourcebuildisfalse__install/lib\"
export testoutofsourcebuildisfalse__man=\"$testoutofsourcebuildisfalse__install/man\"
export testoutofsourcebuildisfalse__doc=\"$testoutofsourcebuildisfalse__install/doc\"
export testoutofsourcebuildisfalse__stublibs=\"$testoutofsourcebuildisfalse__install/stublibs\"
export testoutofsourcebuildisfalse__toplevel=\"$testoutofsourcebuildisfalse__install/toplevel\"
export testoutofsourcebuildisfalse__share=\"$testoutofsourcebuildisfalse__install/share\"
export testoutofsourcebuildisfalse__etc=\"$testoutofsourcebuildisfalse__install/etc\"
"
`;
